src/core/checkSymbolmportability.ts

```diff
@@ -36,15 +36,11 @@ export function checkSymbolImportability(
 
   if (packageOptions.treatSelfReferenceAs === "external") {
     // Check whether this import is the result of a self-reference.
-    const match = possibleSubpathImportFromPackage.exec(moduleSpecifier);
-    if (match) {
-      const packageName = match[1];
-      const lookupResult = lookupPackageJson(importerFilename);
-      if (lookupResult !== null) {
-        if (lookupResult.packageJson.name === packageName) {
-          // This is a self-reference, so treat as external.
-          return;
-        }
+    const lookupResult = lookupPackageJson(importerFilename);
+    if (lookupResult !== null) {
+      if (moduleSpecifier.startsWith(lookupResult.packageJson.name)) {
```

Comment:
:eyes: This might lead to a false negative, e.g. when `lookupResult.packageJson.name === "@uhyo/project"` and `moduleSpecifier === "@uhyo/project2"`.

Also I would like to keep `lookupPackageJson` guarded with the RegExp check in order to reduce file system access.

Reply:
Apologies for ignoring the original intention of regex! I've made the corrections